# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsIContentPolicy.idl
# Commit: 1700d5b79273
# Full Hash: 1700d5b79273a6f34d67b7d16470973f0cb65793
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-21 21:21:23
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement csp for Module Workers; r=evilpie,asuth,rpl,ckerschb
#   
#   Depends on D155691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D156102
# ==============================================================================

diff -r 31888ffde37a -r 1700d5b79273 dom/base/nsIContentPolicy.idl
--- a/dom/base/nsIContentPolicy.idl	Tue Dec 20 20:56:15 2022 +0000
+++ b/dom/base/nsIContentPolicy.idl	Tue Dec 20 20:56:15 2022 +0000
@@ -433,6 +433,20 @@
      */
      TYPE_WEB_IDENTITY = 57,
 
+    /**
+     * Indicates the load of a static module on workers.
+     */
+    TYPE_INTERNAL_WORKER_STATIC_MODULE = 58,
+
+    /**
+     * Used to indicate the end of this list, not a content policy. If you want
+     * to add a new content policy type, place it before this sentinel value
+     * TYPE_END, have it use TYPE_END's current value, and increment TYPE_END by
+     * one. (TYPE_END should always have the highest numerical value.)
+     */
+    TYPE_END = 59,
+
+
     /* When adding new content types, please update
      * NS_CP_ContentTypeName, nsCSPContext, CSP_ContentTypeToDirective,
      * DoContentSecurityChecks, all nsIContentPolicy implementations, the